File: 단계별로 풀어보기/기하 1/4153/직각삼각형.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    
    int dot[3], m, total;
    
    while(cin >> dot[0] >> dot[1] >> dot[2]){
        if(dot[0]==0 && dot[1]==0 && dot[2]==0)
            break;
        
        m = max(dot[0], max(dot[1], dot[2]));
        total = 0;
        for(int i = 0; i < 3; i++){
            if(dot[i] == m)
                continue;
            total += dot[i]*dot[i];
        }
        
        if(m*m == total){
            cout << "right\n";
        } else{
            cout << "wrong\n";
        }
    }

    return 0;
}

File: 단계별로 풀어보기/기하 1/4153/직각삼각형.py
import sys

def init():
    dots = list(map(int,sys.stdin.readline().split()))
    
    while(dots[0] != 0  and  dots[1] != 0 and  dots[2] != 0):
        m = max(dots)
        total = 0
        for i in range(3):
            if dots[i] == m:
                continue
            total += dots[i]* dots[i]
        
        if m*m == total:
            sys.stdout.write('right\n')
        else:
            sys.stdout.write('wrong\n')
            
        dots = list(map(int,sys.stdin.readline().split()))

init()
